Yang HW7 submission

YangJiang89 opened the pull request:
HW 7
@MengyuanZoe @hanghu @jhp312

YangJiang89 committed:
Yang HW7 submission

MengyuanZoe commented on HW7/zipcode_map_Yang.py:
```diff
@@ -0,0 +1,60 @@
+
+import pandas as pd
+import folium
+
```
Add one more blank line.

YangJiang89 replied:
Done

MengyuanZoe commented on HW7/zipcode_map_Yang.py:
```diff
@@ -0,0 +1,60 @@
+    """
+
+    # Load King County house price data
+    house_data = pd.read_csv("../Data/kc_house_data.csv", 
```
Rename the constant name. Avoid underscore. Use "HouseData".

MengyuanZoe commented on HW7/zipcode_map_Yang.py:
```diff
@@ -0,0 +1,60 @@
+    house_data['zipcode'] = house_data['zipcode'].astype(str)
+
+    # Group data by zipcode and calculate the mean value in each zipcode
+    zipcode_data = pd.groupby(house_data, 'zipcode').mean()
```
Rename constant. Use "ZipcodeData".

MengyuanZoe commented on HW7/zipcode_map_Yang.py:
```diff
@@ -0,0 +1,60 @@
+
+    col : string
+        The colomn name in dataset to bound zipcode with
+    """
```
Also discuss the output/return value of this module in the comments.

puleo-joe commented on HW7/zipcode_map_Yang.py:
```diff
@@ -0,0 +1,60 @@
+        The other data set to interact with (house price or crime)
+
+    col : string
+        The colomn name in dataset to bound zipcode with
```
Remove trailing whitespace in lines 25, 29, 30, 31, 43

puleo-joe replied:
spelling error: Change "colomn" to "column"

puleo-joe commented on HW7/zipcode_map_Yang.py:
```diff
@@ -0,0 +1,60 @@
+    # Path for GeoJSON zipcode data
+    zipcodepath = '../Data/zipcode_king_county.geojson'
+    show_zipcode_map(zipcodepath, zipcode_data, 'count')
+    show_zipcode_map(zipcodepath, zipcode_data, 'price')
```
Specify what the expected output for these maps are in a comment.

puleo-joe commented on HW7/zipcode_map_Yang.py:
```diff
@@ -0,0 +1,60 @@
+    zipcode = folium.Map(location=[data['lat'].mean(), 
+                                   data['long'].mean()], zoom_start=10)
+
+    # Add zipcode map layer to orignial map
```
spelling error:  Change "orignial" to "original"

puleo-joe commented on HW7/zipcode_map_Yang.py:
```diff
@@ -0,0 +1,60 @@
+
+if __name__ == "__main__":
+    """
+    Example of using show_zipcode_map funciton
```
Spelling error: Change "funciton" to "function"

YangJiang89 committed:
Update zipcode_map_Yang.py

YangJiang89 committed:
Update zipcode_map_Yang.py

YangJiang89 committed:
Update zipcode_map_Yang.py

MengyuanZoe commented on HW7/zipcode_map_Yang.py:
```diff
@@ -0,0 +1,60 @@
+
+    # Path for GeoJSON zipcode data
+    zipcodepath = '../Data/zipcode_king_county.geojson'
+    show_zipcode_map(zipcodepath, zipcode_data, 'count')
```
Add more comments for two usages of the show_zipcode_map, e.g. what you are plotting.

YangJiang89 committed:
Update zipcode_map_Yang.py

Add more comments on the return of show_zipcode_map function
No changes on variable name zipcode_data and house_data
Add more comment on the the map generated

YangJiang89 commented:
Update my code based on the reviews:
fix the white blank, typos, add more comments
No change on the variable name
